alerting/src/main/kotlin/org/opensearch/alerting/util/AlertingUtils.kt

```diff
@@ -177,3 +186,85 @@ fun ThreadContext.StoredContext.closeFinally(cause: Throwable?) = when (cause) {
         cause.addSuppressed(closeException)
     }
 }
+
+/**
+ * Mustache template supports iterating through a list using a `{{#listVariable}}{{/listVariable}}` block.
+ * https://mustache.github.io/mustache.5.html
+ *
+ * This function looks for `{{#${[AlertContext.SAMPLE_DOCS_FIELD]}}}{{/${[AlertContext.SAMPLE_DOCS_FIELD]}}}` blocks,
+ * and parses the contents for tags, which we interpret as fields within the sample document.
+ *
+ * @return a [Set] of [String]s indicating fields within a document.
+ */
+fun parseSampleDocTags(messageTemplate: Script): Set<String> {
```

Comment:
we should ideally have a parameter `enrichContextWithSampleDocs` in the create monitor request inside the trigger payload

Reply:
Agreed. I'll create an issue to track updating the trigger data model to add this boolean-based logic.

Reply:
Created issue https://github.com/opensearch-project/alerting/issues/1475 to track this action item.